### PR TITLE
Fix issue with subscribing on envs that do not support firebase

### LINKED ISSUE #191: [Brave] Subscribing fails when special Google services setting is not manually enabled
**Describe the bug**

- Subscribing fails in Brave unless you manually enable `Privacy & Security > Use Google services for push messaging` 

**To Reproduce**
1. Open web3inbox
2. Explore > click Subscribe for gm dapp
3. Permission is asked for desktop notifs > allow
4. Instantly shows "Subscribing fails" toast

**Expected behavior**
- Environment concerns around desktop notifications do not block functionality/subscribing to dapps

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Brave

### PR DESCRIPTION
**Pull request by devceline**
# Description

- Wrap firebase logic in try/catch
- Tested on brave with & without google notification permissions

# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Fixes/Resolves (Optional)

Resolves #191 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/w3iProxy/pushProviders/internalPushProvider.ts [hunk 1]
```diff
@@ -136,27 +136,31 @@ export default class InternalPushProvider implements W3iPushProvider {
       const clientId = await this.pushClient.core.crypto.getClientId()
       // Retrieving FCM token needs to be client side, outside the service worker.
```
Comment (bkrem) [inline src/w3iProxy/pushProviders/internalPushProvider.ts:137]: Comment is now out of place, needs to be moved

File: src/w3iProxy/pushProviders/internalPushProvider.ts [hunk 2]
```diff
@@ -136,27 +136,31 @@ export default class InternalPushProvider implements W3iPushProvider {
...
-          this.pushClient?.off('push_subscription', subEvListener)
+            this.pushClient?.off('push_subscription', subEvListener)
+          }
         }
-      }
 
-      this.pushClient.on('push_subscription', subEvListener)
+        this.pushClient.on('push_subscription', subEvListener)
+      } catch (e) {
+        console.error(e)
```
Comment (bkrem) [inline src/w3iProxy/pushProviders/internalPushProvider.ts:162]: Please add a proper error message to prefix the error so we can easily identify it at a later point without needing to dig through stack traces/set breakpoints.

**Review by bkrem — APPROVED** [review]

File: src/w3iProxy/pushProviders/internalPushProvider.ts (hunk 1 above)
Comment (devceline) [reply src/w3iProxy/pushProviders/internalPushProvider.ts]: Moved comment

File: src/w3iProxy/pushProviders/internalPushProvider.ts (hunk 2 above)
Comment (devceline) [reply src/w3iProxy/pushProviders/internalPushProvider.ts]: Added prefix

**Pull request merged**